### PR TITLE
#167165021 Users should be able to view their profiles

### PR DESCRIPTION
**Pull request by tolulope-od**
### What does this PR do?
Creates a profile card component
Creates profile view with mock data
Add a profile component route
Writes tests for newly added profile card and profile tab component

#### Description of Task to be completed?
A user should have access to their profile and view and edit details on the profile page by navigating to `/profile`

#### How should this be manually tested?
After cloning this repo, `cd` into it and do the following:

```bash
# fetch this branch
$ git fetch origin ft-edit-and-view-profile-167165021

# install dependencies
$ npm install

# unit and integration tests are available by running this script
$ npm test

# start the development server
$ npm run dev
```

After doing this, you should see a message appear on the console, with the app running on Port 8080, you can view the app on `localhost:8080`. You can view the app on the above stated route

#### Any background context you want to provide?
N/a

#### What are the relevant pivotal tracker stories?
[#167165021](https://www.pivotaltracker.com/story/show/167165021)

#### Screenshots (if appropriate)
Screenshot 2019-08-28 at 7 51 34 PM

### CODE REVIEW

File: src/components/common/tabs/ProfileTab.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/* eslint-disable max-len */
```
Comment (mazma1) [inline src/components/common/tabs/ProfileTab.js]: Pls fix any max-length error, this shouldn't be a big deal

File: src/components/common/tabs/ProfileTab.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      <div id="stories" className="tabcontent">
+        <div className="profileStories">
+          { profile.loading && <Loader /> }
+          { !profile.loading && profile.articles.length === 0 && <h3>No Articles To Display At This Time</h3> }
+          { !profile.loading && profile.articles.length > 0 && profile.articles.map((article, key) => article.status === 'published' && <div className="singleStory" key={key.toString()}><Card {...article} key={article.id} /></div>) }
+        </div>
+      </div>
+      <div id="bookmarks" className="tabcontent">
+        <h3>Bookmarks</h3>
+        <p>Bookmarks here</p>
```
Comment (mazma1) [inline src/components/common/tabs/ProfileTab.js]: Can you update the message displayed when there is no bookmark?

File: src/components/common/tabs/ProfileTab.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/* eslint-disable max-len */
+
+import React, { useEffect } from 'react';
+import { withRouter } from 'react-router-dom';
+import Card from '@Common/landingPage/card/Cards';
+import Loader from '@Common/loader/Loader';
+import './profileTab.scss';
+
+const ProfileTab = (profile) => {
+  const openTab = (e, tabName) => {
```
Comment (mazma1) [inline src/components/common/tabs/ProfileTab.js]: No `e` variable name pls

File: src/components/common/tabs/ProfileTab.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          </>
+        )
+        }
+      </div>
+      <form>
+        <div className="search_input_wrap">
+          <i className="material-icons" id="search__icon">search</i>
+          <input type="search" className="browser-default" id="search__profile" placeholder="Seach Here" />
+        </div>
+      </form>
```
Comment (mazma1) [inline src/components/common/tabs/ProfileTab.js]: Can this search form be displayed when we have like a given number of content? It looks off when there is nothing to search

**Comment by mazma1** [comment]
@tolulope-od great work getting this to a point where all the checks are clear. 💪🏽
Just to be sure, the edit feature is not implemented yet, right?

**Comment by tolulope-od** [comment]
> @tolulope-od great work getting this to a point where all the checks are clear. 💪🏽
> Just to be sure, the edit feature is not implemented yet, right?

Thank you.
Yes the edit feature is not yet implemented

File: src/components/common/tabs/ProfileTab.js (hunk 4 above)
Comment (tolulope-od) [reply src/components/common/tabs/ProfileTab.js]: I'll implement this. Thank you for the suggestion

**Pull request merged**